src/navigation/checkout/Restaurant.js

```diff
@@ -46,121 +64,226 @@ const LoadingPhantom = (props) =>
     </VStack>
     <Skeleton flex="1" h="100" w="100" rounded="md" />
   </HStack>
+);
 
 function Restaurant(props) {
-  const { navigate } = props.navigation
-  const colorScheme = useColorScheme()
-  const [ infoModal, setInfoModal ] = useState(false)
-  const { showFooter, httpClient, restaurant, openingHoursSpecification } = props
+  const {navigate} = props.navigation;
+  const colorScheme = useColorScheme();
+  const [infoModal, setInfoModal] = useState(false);
+  const {showFooter, httpClient, restaurant, openingHoursSpecification} = props;
+  const sectionListRef = useRef(null);
+
+  const [activeSection, setActiveSection] = useState(null);
+  const viewabilityConfig = useRef({viewAreaCoveragePercentThreshold: 50});
+  const handleViewableItemsChanged = useRef(({viewableItems}) => {
+    if (viewableItems.length > 0) {
+      setActiveSection(viewableItems[0].index);
+    }
+  });
+  const screenHeight = Dimensions.get('window').height;
 
-  const { isLoading, isError, data } = useQuery([ 'menus', restaurant.hasMenu ], async () => {
-    return await httpClient.get(restaurant.hasMenu, { anonymous: true })
-  })
+  const {isLoading, isError, data} = useQuery(
+    ['menus', restaurant.hasMenu],
+    async () => {
+      return await httpClient.get(restaurant.hasMenu, {}, {anonymous: true});
+    },
+  );
 
   const currentTimeSlot = useMemo(
     () => openingHoursSpecification.currentTimeSlot,
-    [openingHoursSpecification]
-  )
+    [openingHoursSpecification],
+  );
 
   //TODO: improve failed view
   if (isError) {
-    return <Center w="95%">
-      <Heading>{i18n.t('AN_ERROR_OCCURRED')} </Heading>
-      <Text>{i18n.t('TRY_LATER')}</Text>
-    </Center>
+    return (
+      <Center w="95%">
+        <Heading>{i18n.t('AN_ERROR_OCCURRED')} </Heading>
+        <Text>{i18n.t('TRY_LATER')}</Text>
+      </Center>
+    );
   }
 
   const cardStyle = {
     ...styles.card,
-    backgroundColor: colorScheme === 'dark' ? 'rgba(255, 255, 255, 0.04)' : 'rgba(0, 0, 0, 0.02)',
-    borderColor: colorScheme === 'dark' ? 'rgba(255, 255, 255, 0.2)' : 'rgba(0, 0, 0, 0.1)',
-  }
+    backgroundColor:
```

Comment:
maybe better to put these colors into global `styles.js`? I think we'll need them somewhere else

Reply:
Yes, that's what I was going to do once we'd determined the official colors for coopcycle, because right now I'm using probably temporary colors that I found in the figma
I'm trying to split the PL goals as much as possible to have several small PL rather than one big one